### PR TITLE
replace use of unmaintained setup-scala action

### PR DESCRIPTION
**Pull request by pjfanning**
related to https://github.com/apache/incubator-pekko-connectors-kafka/pull/26

### CODE REVIEW

File: .github/workflows/format.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: Scalafmt
 
-permissions: read-all
+permissions: {}
```
Comment (pjfanning) [inline .github/workflows/format.yml:3]: I don't think read-all is needed. `{}` disables the jobs access to repository secrets - and this should be the go-to setting unless a job needs to read secrets

Comment (pjfanning) [reply .github/workflows/format.yml:3]: https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs

Comment (mdedetrich) [reply .github/workflows/format.yml:3]: Are you sure this is what you intend? `read-all` is more strict than `{}`, not less.

Comment (mdedetrich) [reply .github/workflows/format.yml:3]: Are you sure this is what you intend? `read-all` is more strict than `{}`, not less.

Comment (mdedetrich) [reply .github/workflows/format.yml:3]: Thanks for clarifying, had a misunderstanding then

**Review by mdedetrich — APPROVED** [review]
Lgtm

File: .github/workflows/format.yml (hunk 1 above)
Comment (pjfanning) [reply .github/workflows/format.yml:3]: To be frank, the Github docs are not well written - leaving plenty of ambiguity.

I'm not sure why not setting permissions at all in the yml does not default to having the lowest available permissions.

A few months ago, there were people popping PRs on dozens of repos to set `permissions: {}` but honestly, they could have been wrong.

**Pull request merged**